### PR TITLE
First attempt

### PR DESCRIPTION
**Pull request by BearintheWillows**

### CODE REVIEW

File: Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+using System.Diagnostics;
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs]: Remove unused using

File: Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+[ApiController]
+public class TodosController : ControllerBase
+{
+    private readonly TodoDbContext _context;
+
+    public TodosController(TodoDbContext context)
+    {
+        _context = context;
+    }
+
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs:19]: Remove empty lines 20,21

File: Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<TodoItem>>> GetTodoItems()
+    {
+        return await _context.TodoItems.OrderBy(x => x.IsDone).ToListAsync();
+
+    }
+
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs:26]: Remove empty line 29

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            modelBuilder.ApplyConfiguration(new TodoItemConfiguration());
+        }
+        public DbSet<TodoItem> TodoItems { get; set; }
+
+        //Add Todo
+        /// <summary>
+        /// Adds param to Database table
+        /// </summary>
+        /// <param name="todoItem"></param>
+        public void AddTodoItem(TodoItem todoItem)
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Please remove all methods that aren't being used 😁

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+using Serilog;
+
+namespace Todo.WillowBear.Data
+{
+    public class TodoDbContext : DbContext
+    {
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlite("Data Source=TodoList.db");
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs:10]: As I understand (and I didn't make it clear in the requirements), you created the tables using as script directly against the DB, correct? If that's the case could you please send me the script you used to create the DB?

Thanks!

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs [hunk 6]
```diff
@@ -0,0 +1,160 @@
+using Microsoft.EntityFrameworkCore;
+using Todo.WillowBear.Models;
+using Serilog;
+
+namespace Todo.WillowBear.Data
+{
+    public class TodoDbContext : DbContext
+    {
+
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Empty line

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        /// </summary>
+        /// <param name="todoItem"></param>
+        public void AddTodoItem(TodoItem todoItem)
+        {
+
+            if (todoItem == null)
+            {
+                Log.Debug($"{todoItem} is null");
+            }
+
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Empty line

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+            try
+            {
+                TodoItems.Update(todoItem!);
+                SaveChanges();
+                Log.Information($"Todo Item {todoItem!.Id} has been updated successfully");
+            }
+            catch (Exception e)
+            {
+
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Empty line

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+            }
+            catch (Exception e)
+            {
+                Log.Information($"Unable to retrieve Todo Item. ");
+                Log.Debug(e.Message);
+                return null;
+            }
+        }
+
+        // get all todoitems
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: It's ok to add these comments for personal reference but your method signature should be self-explanatory. In most companies, you wouldn't push these types of comments to a team.

File: Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs [hunk 10]
```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs]: Remove unused usings

File: Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Todo.WillowBear.Models;
+
+namespace Todo.WillowBear.Data
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs]: Use new namespaces

File: Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs [hunk 12]
```diff
@@ -0,0 +1,14 @@
+namespace Todo.WillowBear.Models
+{
+
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs]: Empty line

File: Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+    public class TodoItem
+    {
+        public int Id { get; set; }
+        public string Title { get; set; } = string.Empty;
+        public int IsDone { get; set; }
+
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs]: It's ok to separate properties in groups with empty lines, but what's the criteria here?

File: Todo.WillowBear/Todo.WillowBear/Program.cs [hunk 14]
```diff
@@ -0,0 +1,55 @@
+using Serilog;
+using Todo.WillowBear.Data;
+;
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Program.cs]: Remove semi-colon and empty line

File: Todo.WillowBear/Todo.WillowBear/Program.cs [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+
+
+//add serilog
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Program.cs]: Unnecessary comment

File: Todo.WillowBear/Todo.WillowBear/Program.cs [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+    .WriteTo.Console()
+    .CreateLogger();
+
+try
+{
+    Log.Information("Starting web application");
+
+    var builder = WebApplication.CreateBuilder(args);
+
+    // Add services to the container.
```
Comment (TheCSharpAcademy) [inline Todo.WillowBear/Todo.WillowBear/Program.cs]: Again, comments are ok for learning, but not for pushing to production

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]

File: Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs (hunk 11 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs]: Could you clarify this one for me please? Should it be in its own namespacE?

File: Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs (hunk 13 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs]: Un-intended! Have fixed.

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs (hunk 5 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs:10]: I didn't! The following code in TodoItemsConfiguration.cs created the table

public class TodoItemConfiguration : IEntityTypeConfiguration<TodoItem>
    {
        public void Configure(EntityTypeBuilder<TodoItem> builder)
        {
            builder.ToTable("TodoItems");
            builder.HasKey(x => x.Id);
            builder.Property(x => x.Title).IsRequired().HasMaxLength(20);
            builder.Property(x => x.IsDone).IsRequired().HasDefaultValue(0);
            builder.Property(x => x.CreatedAt).IsRequired().HasDefaultValueSql("datetime('now')");
            builder.Property(x => x.DueDate);
        }
    }

Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs:10]: Should I remove that file and by directly against the Db do you mean an SQL script within SQL management studio?

File: Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs (hunk 1 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs (hunk 2 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs:19]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs (hunk 3 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Controllers/TodosController.cs:26]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs (hunk 4 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs (hunk 6 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs (hunk 7 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs (hunk 8 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs (hunk 9 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs]: Fixed.

Can you advise on a good resource or explain what comments should be left for production commits?

File: Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs (hunk 10 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Data/TodoItemConfiguration.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs (hunk 12 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Models/TodoItem.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Program.cs (hunk 14 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Program.cs]: Fixed.

File: Todo.WillowBear/Todo.WillowBear/Program.cs (hunk 15 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Program.cs]: Fixed

File: Todo.WillowBear/Todo.WillowBear/Program.cs (hunk 16 above)
Comment (BearintheWillows) [reply Todo.WillowBear/Todo.WillowBear/Program.cs]: Fixed.

**Pull request closed without merging**